src/hmm/hmm-utils.cc

```diff
@@ -49,8 +49,8 @@ fst::VectorFst<fst::StdArc> *GetHmmAsFst(
   const HmmTopology &topo = trans_model.GetTopo();
   const HmmTopology::TopologyEntry &entry  = topo.TopologyForPhone(phone);
 
-  // vector of the pdfs, indexed by pdf-class (pdf-classes must start from zero
-  // and be contiguous).
+  // vector of the pdfs, indexed by pdf-class and self-loop pdf-class
```

Comment:
you can revert this change in wording.  pdf-class and self-loop pdf-class are not different types of index: self-loop pdf-class is the pdf-class that is on the self-loop.